common/flatpak-run.c

```diff
@@ -407,6 +408,42 @@ flatpak_run_add_environment_args (FlatpakBwrap    *bwrap,
             }
         }
 
+      if (context->devices & FLATPAK_CONTEXT_DEVICE_INPUT)
+        {
+          const gchar *sysfs_path = "/sys/class/hidraw";
+          g_autoptr(GDir) sysfs_dir = NULL;
+          g_autoptr(GError) dir_error = NULL;
+
+          g_debug ("Allowing input device access");
+
+          /* Enumerate hidraw device nodes by checking /dev/ for names that
+           * match the symlink names in /sys/class/hidraw/ */
+          sysfs_dir = g_dir_open (sysfs_path, 0, &dir_error);
+          if (sysfs_dir == NULL)
+              g_warning ("Cannot read %s: %s", sysfs_path, dir_error->message);
+          else
+            {
+              gchar dev_path[sizeof ("/dev/hidraw") + 3];
+              const gchar *name;
+              while ((name = g_dir_read_name (sysfs_dir)) != NULL)
+                {
+                  if (sizeof ("/dev/") + strlen (name) > sizeof (dev_path))
+                    {
+                      g_warning ("Skipping unexpectedly long hidraw name: %s", name);
+                      continue;
+                    }
+                  g_snprintf (dev_path, sizeof (dev_path), "/dev/%s", name);
+                  if (g_access (dev_path, R_OK) == 0)
+                      flatpak_bwrap_add_args (bwrap, "--dev-bind", dev_path, dev_path, NULL);
+                }
+            }
+
+          if (g_file_test ("/dev/uinput", G_FILE_TEST_EXISTS))
+              flatpak_bwrap_add_args (bwrap, "--dev-bind", "/dev/uinput", "/dev/uinput", NULL);
```

Comment:
By default (on systems that don't have steam-devices or other custom udev rules), `/dev/uinput` is not accessible by ordinary unprivileged users. Does this PR still work on systems where `/dev/uinput` exists but cannot be opened?

Reply:
Wouldn't this just error out when a program inside of Flatpak tries to access `/dev/uinput`? Same idea as binding all of `/dev`, I think.

Reply:
If you are interested in bringing this PR closer to being merged, perhaps you could try it and find out?

Reply:
Normally, unprivileged users can open the `/dev` directory, even if they cannot open individual device nodes within `/dev`: so no, it isn't the same. Bind-mounting all of `/dev` doesn't require Flatpak/bubblewrap to open every device node within `/dev` individually.